### ISSUE TITLE
Auto completion does not work in a dplyr pipe when the piped object is not directly defined in the environment

### ISSUE THREAD

**Issue by amitramon** [rstudio/rstudio#9612]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.1717
    OS Version      : Debian Gnu/Linux
    R Version       : 4.04

### Steps to reproduce the problem

1. Evaluate the following code

```
library(tidyverse)

outer_lst <- list(
  lvl1_lst =list(
    inner_df = data.frame(inner_col_1 = 1:2, inner_col_2 = 3:4),
    inneer_x = 42
  ),
  lvl1_df = data.frame(lvl1_col_1 = 1:2, lvl1_col_2 = 3:4)
)
```
2. Try to get auto-completions by placing the cursor inside the filter() parenthesis in each of the following examples and pressing TAB.
```
outer_lst$lvl1_lst$inner_df %>% filter()

outer_lst$lvl1_lst[[1]] %>% filter()

outer_lst$lvl1_df %>% filter()

outer_lst[[2]] %>% filter()
```
Problem: the column names for the `data.frame` that is piped into `filter()` are not shown in the auto-complete list.

### Describe the problem in detail

Piping a `data.frame` object into a `dplyr` pipe. If the `data.frame` object is directly in the environment, as in

```
df <- data.frame(col_1 = 1:2, col_2 = 3:4)
```
then, when piping `df` into a pipe
```
df %>% filter()
```
if I place the cursor inside the parenthesis of the `filter()` function, pressing TAB will show the columns of `df` in the completions list - this is the expected behavior.

But if `df` is defined inside another object, say a `list`, then the columns will not be shown in the completions list:

```
a_list <- list(df = data.frame(col_1 = 1:2, col_2 = 3:4))
## auto-completions in the filter() function do not include the columns of df
a_list$df %>% filter()
```

### Describe the behavior you expected

When a `data.frame` object is piped into a function (e.g. `filter()` from the `dplyr` package, as in the above examples), I expect that the list generated by the auto-completion mechanism will include the column names of the object's columns.
However, this works when the object is defined directly in the environment, but it does not work if the object is defined inside another object. 

### Notes

I used the inspection tool (right-click in Rstudio, select 'Inspect element') for inspecting the request sent from the Rstudio UI client to the associated R session. It seems that the request's payload does not contain the fully-qualified name of the 'chain object'. So even if the object on the LHS of the pipe is `a_list$df`, the payload only contains `df` for the chain object.

Mimicking a request using `curl`, and modifying the chain object to be the fully-qualified name of the piped object, I'm managing to get the column names correctly.  

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by amitramon** [comment]
UI examples.

1. object is defined directly in the environment, auto-completion for column names works:
![pipe2s](https://user-images.githubusercontent.com/9901460/126031919-ee0efa50-b150-44aa-a3f1-5b1b1604f2c7.png)

2. object is defined inside another object, auto-completion does not show the object's column names:
![pipe1s](https://user-images.githubusercontent.com/9901460/126031909-5475cf6f-adca-4a19-93a3-b92c0659415c.png)

**Comment by ronblum** [comment]
@amitramon Thank you for raising this! I can also reproduce this in
- RStudio Desktop 1.3.1093 on MacOS 12 beta 
- RStudio Desktop 1.5.233 on MacOS 12 beta
- RStudio Server 1.5.192 on SLES 15 SP2 via Safari on MacOS 12 beta

I'll mark this for review.

**Comment by liyezhou** [comment]
Same issue!

**Comment by romainfrancois** [comment]
The internals of the completion system get `inner_df` as the `chainObjectName`. 

```r
.rs.rpc.get_completions(token = "", string = list("filter"), 
    context = list(1L), numCommas = list(0L), functionCallString = "filter(", 
    chainObjectName = "inner_df", additionalArgs = list(), excludeArgs = list(), 
    excludeArgsFromObject = FALSE, filePath = "", documentId = "", 
    line = "outer_lst$lvl1_lst$inner_df %>% filter(", isConsole = TRUE)
```

instead of `outer_lst$lvl1_lst$inner_df`

**Comment by yuhenghuang** [comment]
I am having similar problems and have come to see this issue and the pull requst above.
![image](https://user-images.githubusercontent.com/41667928/208301689-c95e8b8a-bd99-4000-8d74-2bf549340562.png)

The environment is
> Ubuntu 22.04 (x86_64), kernel version 5.15.0-56-generic
> R version 4.2.2 Patched (2022-11-10 r83330) -- "Innocent and Trusting"
> "Elsbeth Geranium" Release (7d165dcf, 2022-12-03) for Ubuntu Jammy

The newest version of RStudio doesn't seem to have updated this part.

Does anyone know when this issue will be solved in the official release, or is there anything I can do to cope with it?

Thanks!

**Comment by romainfrancois** [comment]
This appears fixed in build RStudio 2023.03.0-daily+34 "Cherry Blossom" Daily (8ae98d75012c643f87e57cd625100914e9377aaa, 2022-12-21) for macOS
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.03.0-daily+34 Chrome/108.0.5359.62 Electron/22.0.0 Safari/537.36

<img width="621" alt="image" src="https://user-images.githubusercontent.com/2625526/210321744-1a935589-ac44-45b1-a9d4-3a4ba44092c5.png">

This will be released as Cherry Blossom, planned for march 2023.
